### PR TITLE
Fix boolean logic with regards to spark secure connections

### PR DESCRIPTION
**Pull request by cccs-jc**
**Description of the change**

Fix issue with enabling secure connection and storage. The template did use boolean values which is good practice.

Also added a tls entry for the ingress template

### CODE REVIEW

File: bitnami/spark/values-production.yaml [hunk 1]
```diff
@@ -243,18 +243,18 @@ security:
   ## RPC configuration
   ##
   rpc:
-    authenticationEnabled: "yes"
-    encryptionEnabled: "yes"
+    authenticationEnabled: true
+    encryptionEnabled: true
```
Comment (sameersbn) [inline bitnami/spark/values-production.yaml]: These changes probably require a bump in the  major version. wdyt @javsalgar @juan131

**Comment by cccs-jc** [comment]
this pull request is ready to be merged. Please let me know if there's anything else to fix.

File: bitnami/spark/templates/statefulset-master.yaml [hunk 2]
```diff
@@ -71,28 +71,27 @@ spec:
             - name: SPARK_MASTER_OPTS
               value: {{ .Values.master.configOptions | quote }}
             {{- end }}
-            {{- if eq ( .Values.security.rpc.authenticationEnabled | quote ) "yes" }}
+            {{- if .Values.security.rpc.authenticationEnabled }}
```
Comment (sameersbn) [inline bitnami/spark/templates/statefulset-master.yaml]: The spark container expects `yes/no` for boolean variables. Changing it here would also need the spark container to work with `true/false` as values for boolean variables. Could you please describe the issue this PR attempts to fix? Please share the issue number if there is an associated issue. Or possible share details how to reproduce the issue this PR attempts to fix

File: bitnami/spark/templates/statefulset-master.yaml [hunk 3]
```diff
@@ -71,28 +71,27 @@ spec:
...
                   name: {{ include "spark.fullname" . }}-secret
-                {{- end }}
                   key: rpc-authentication-secret
             {{- end }}
-            {{- if eq ( .Values.security.encryptionEnabled | quote ) "yes" }}
+            {{- if .Values.security.rpc.encryptionEnabled }}
             - name: SPARK_RPC_ENCRYPTION_ENABLED
               value: {{ .Values.security.rpc.encryptionEnabled | quote }}
             {{- end }}
-            {{- if eq ( .Values.security.storageEncriptionEnabled | quote ) "yes" }}
```
Comment (sameersbn) [inline bitnami/spark/templates/statefulset-master.yaml]: I think the fix to the storage issue you are trying to fix is to correct this typo.

File: bitnami/spark/templates/statefulset-master.yaml (hunk 2 above)
Comment (cccs-jc) [reply bitnami/spark/templates/statefulset-master.yaml]: The container works with true/false. The use of "yes" and "no" is bad practice in Helm, this is a boolean value and should be true/false. Other charts in bitnami uses true/false.

Comment (cccs-jc) [reply bitnami/spark/templates/statefulset-master.yaml]: Also had issue with the version of Helm I was using. If I recall correctly it was interpretting yes/no as booleans. Which did not work with the string compare test done in the chart. The values "yes" and "no" needed to be quoted or else they would be interpreted as boolean and were not processed properly by the chart. It's just a meassy way to specify boolean values which should be true/false.

Comment (sameersbn) [reply bitnami/spark/templates/statefulset-master.yaml]: @cccs-jc thanks for the information.

File: bitnami/spark/templates/statefulset-worker.yaml [hunk 4]
```diff
@@ -92,29 +92,24 @@ spec:
...
-                {{- if .Values.security.passwordsSecretName }}
                   name: {{ .Values.security.passwordsSecretName }}
                   key: rpc-authentication-secret
-                {{- else }}
-                  name: {{ include "spark.fullname" . }}-secret
-                  key: rpc-authentication-secret
-                {{- end }}
             {{- end }}
-            {{- if eq ( .Values.security.encryptionEnabled | quote ) "yes" }}
+            {{- if .Values.security.rpc.encryptionEnabled }}
```
[the remark this answers is not in the document]
Comment (cccs-jc) [reply bitnami/spark/templates/statefulset-worker.yaml]: fixed

**Comment by sameersbn** [comment]
@cccs-jc I will test the changes before merging. You're contribution is much appreciated. Thank you!

File: bitnami/spark/templates/statefulset-worker.yaml [hunk 5]
```diff
@@ -123,19 +118,19 @@ spec:
                   name: {{ template "spark.get.passwordSecretName" }}
```
Comment (sameersbn) [inline bitnami/spark/templates/statefulset-worker.yaml]:
```suggestion
                  name: {{ template "spark.get.passwordSecretName" . }}
```

File: bitnami/spark/templates/statefulset-worker.yaml [hunk 6]
```diff
@@ -123,19 +118,19 @@ spec:
...
                   key: ssl-key-password
             - name: SPARK_SSL_KEYSTORE_PASSWORD
-                valueFrom:
-                  secretKeyRef:
-                    name: {{ template "spark.get.passwordSecretName" }}
-                    key: ssl-keystore-password
+              valueFrom:
+                secretKeyRef:
+                  name: {{ template "spark.get.passwordSecretName" }}
```
[the remark this answers is not in the document]
Comment (sameersbn) [reply bitnami/spark/templates/statefulset-worker.yaml]: you missed this one

File: bitnami/spark/templates/statefulset-worker.yaml [hunk 7]
```diff
@@ -123,19 +118,19 @@ spec:
...
+                  key: ssl-keystore-password
             - name: SPARK_SSL_TRUSTSTORE_PASSWORD
-                valueFrom:
-                  secretKeyRef:
-                    name: {{ template "spark.get.passwordSecretName" }}
-                    key: ssl-truststore-password
+              valueFrom:
+                secretKeyRef:
+                  name: {{ template "spark.get.passwordSecretName" }}
```
[the remark this answers is not in the document]
Comment (sameersbn) [reply bitnami/spark/templates/statefulset-worker.yaml]: you missed this one as well

File: bitnami/spark/templates/statefulset-master.yaml [hunk 8]
```diff
@@ -71,49 +71,49 @@ spec:
...
                 secretKeyRef:
-                {{- if .Values.security.passwordsSecretName }}
+                  {{- if .Values.security.passwordsSecretName }}
                   name: {{ .Values.security.passwordsSecretName }}
-                {{- else }}
+                  {{- else }}
                   name: {{ include "spark.fullname" . }}-secret
-                {{- end }}
                   key: rpc-authentication-secret
+                  {{- end }}
```
Comment (sameersbn) [inline bitnami/spark/templates/statefulset-master.yaml]: wrong location. It should be before `key: rpc-authentication-secret`

File: bitnami/spark/templates/statefulset-worker.yaml (hunk 6 above)
[the remark this answers is not in the document]
Comment (sameersbn) [reply bitnami/spark/templates/statefulset-worker.yaml]: fixed!

File: bitnami/spark/templates/statefulset-worker.yaml (hunk 5 above)
Comment (sameersbn) [reply bitnami/spark/templates/statefulset-worker.yaml]: fixed!

File: bitnami/spark/templates/statefulset-master.yaml (hunk 8 above)
Comment (sameersbn) [reply bitnami/spark/templates/statefulset-master.yaml]: fixed!

**Comment by sameersbn** [comment]
Thanks for the contribution. Please let us know if you encounter any issues with the chart.

**Pull request merged**